extensions/amp-viewer-integration/0.1/test/integration/test-amp-viewer-integration.js

```diff
@@ -230,6 +231,79 @@ describes.realWin(
             expect(initFocusHandlerStub).to.be.called;
           });
       });
+
+      // TODO(dmanek): remove `ifChrome` once other major browsers support
+      // text fragments (i.e. 'fragmentDirective' in document = true)
+      it.configure()
+        .ifChrome()
+        .run(
+          'should update url based on text fragment post message',
+          async () => {
+            toggleExperiment(win, 'enable-text-fragments', true, true);
+
+            const updateUrlWithTextFragmentSpy = env.sandbox.spy();
+            ampViewerIntegration.updateUrlWithTextFragment_ = updateUrlWithTextFragmentSpy;
+
+            const text = 'will this higlight?';
+            const message = {
+              directive: text,
+            };
+
+            const mockEvent = new CustomEvent('message');
+            mockEvent.origin = 'https://www.google.com';
+            mockEvent.data = message;
+
+            ampViewerIntegration.init();
+
+            win.dispatchEvent(mockEvent);
+            expect(updateUrlWithTextFragmentSpy.calledOnce).to.be.true;
+            expect(updateUrlWithTextFragmentSpy.getCall(0).args[0]).to.equal(
+              text
+            );
+          }
+        );
+
+      it.configure()
+        .ifChrome()
+        .run('should not update url if origin it not google.com', async () => {
+          toggleExperiment(win, 'enable-text-fragments', true, true);
+
+          const updateUrlWithTextFragmentSpy = env.sandbox.spy();
+          ampViewerIntegration.updateUrlWithTextFragment_ = updateUrlWithTextFragmentSpy;
+
+          ampViewerIntegration.init();
+
+          const message = {
+            directive: 'will this highglight?',
+          };
+          const mockEvent = new CustomEvent('message');
+          mockEvent.origin = 'https://www.goøgle.com';
```

Comment:
Tricky! I didnt catch this url as being invalid with a quick glance. Can we use a more clearly invalid value?

`https://not.valid.domain` or similar?

Reply:
That was the intentions 🙂  But on second thoughts, tests should have clear pass/fail, so updated this.